src/study.js

```diff
@@ -1,12 +1,46 @@
+import Glean from "@mozilla/glean/webext";
+import PingEncryptionPlugin from "@mozilla/glean/webext/plugins/encryption";
+
 import { Rally, runStates } from "@mozilla/rally";
 import { onPageData, stopMeasurement } from "./attention-reporter";
-  
+
+import * as pageMetrics from "../src/generated/page.js";
+import * as pageVisitMetrics from "../src/generated/pageVisit.js";
+import * as eventMetrics from "../src/generated/event.js";
+import * as pageAudioMetrics from "../src/generated/pageAudio.js";
+import * as rs01Pings from "../src/generated/pings.js";
+
 function collectEventDataAndSubmit(rally, devMode) {
   // note: onPageData calls startMeasurement.
   onPageData.addListener(async (data) => {
     if (devMode) {
       console.debug("RS01.event", data);
     }
+
+    // Report the data using Glean.
+    pageMetrics.id.set(data.pageId);
+    pageMetrics.origin.set(data.origin);
+    pageMetrics.referrerOrigin.set(data.referrerOrigin);
+    pageVisitMetrics.start.set(new Date(data.pageVisitStartTime));
+    pageVisitMetrics.stop.set(new Date(data.pageVisitStopTime));
+    eventMetrics.start.set(new Date(data.eventStartTime));
+    eventMetrics.stop.set(new Date(data.eventStopTime));
+    // `setRawNanos` expects the input to be in nanoseconds so
+    // convert it before setting.
+    eventMetrics.duration.setRawNanos(data.duration * 1000000);
+    eventMetrics.terminationReason.set(data.eventTerminationReason);
+    pageMetrics.title.set(data.title);
+    pageMetrics.ogType.set(data.ogType);
+    pageMetrics.ogDescription.set(data.description);
+
+    if (data.eventType === "audio") {
+      pageAudioMetrics.maxRelativeScrollDepth.set(data.maxRelativeScrollDepth);
```

Comment:
just a note that given the suggestions in this file, these will have to change to (for example) `pageAttentionMetrics.maxRelativeScrollDepth.set(...)`